### ISSUE TITLE
Update next / previous item navigation field, to use items according to current filtering

### ISSUE THREAD

**Issue by Lyquix** [FLEXIcontent/flexicontent-cck#105]
I wanted to check what you think of a new item field that generates a link for the next (or previous) item.

This is very useful for blogs or any other item type where there is some continuity among items.

We have implemented this functionality in blog sections for two projects, you can see examples here:

http://www.destinationimagination.org/blog/destination-imagination-headquarters-closed-for-the-holidays
http://www.f1plus.com/en/news/item/7189-ferrari-sticking-with-pullrod-in-2015-report

And we have use a code like this in the item.php template:

````php
// get items
$db = JFactory::getDBO();
$query = "SELECT c.id, c.title, c.alias FROM #__content as c, #__flexicontent_cats_item_relations as fcir WHERE c.id = fcir.itemid AND c.state = 1 AND c.publish_up < '".$now."' AND (c.publish_down = '0000-00-00 00:00:00' OR c.publish_down > '".$now."') AND fcir.catid = ".$catid." ORDER BY c.created";
$db->setQuery($query);
$catitems = $db->loadAssocList();
for($i=0; $i<count($catitems); $i++) if($catitems[$i]['id']==$this->item->id) break;
echo '<ul class="prev-next">';
// is there a previous item?
if($i>0) {
	$url = JRoute::_('index.php?option=com_flexicontent&view=item&id='.$catitems[$i-1]['id'].':'.$catitems[$i-1]['alias'].'&Itemid='.$menuitem);
	echo '<li class="prev"><a href="'.$url.'">&laquo; Previous Article</a></li>';
}
// is there a next item
if($i<count($catitems)-1) {
	$url = JRoute::_('index.php?option=com_flexicontent&view=item&id='.$catitems[$i+1]['id'].':'.$catitems[$i+1]['alias'].'&Itemid='.$menuitem);
	echo '<li class="next"><a href="'.$url.'">Next Article &raquo;</a></li>';
}
````

I doubt this is the most efficient or elegant code, but it works fine. As you can see the logic is:

a) Execute a query that will result in a list of ordered items
b) Find if there is a previous item and generate a link
c) Find if there is a next item and generate a link

For a field that executes this function I think we would need the following:

1. Define scope to generate list of items
  * Categories (all, include, exclude)
  * Item Type (all, include, exclude)
  * Same language (yes/no)
  * Ordering field (alpha, date, ID, ordering, hits, custom field) and ordering direction
2. Render links for Previous only, Next only, or both
3. Labels
  * Custom text (e.g. "Next Article", "Previous Project")
  * Item title
  * Custom item field

Let me know what you think of this. If you think this is useful, we can added to our initiatives for 2015.

**Comment by micker** [comment]
Field navigation done it 
You dont like it ?

**Comment by Lyquix** [comment]
Oh, I am so sorry, I didn't see it before!
I will check it and let you know if there are any improvements I can suggest.
Thank you!

**Comment by micker** [comment]
you have :
next and previous link
categorie link return
tooltip function
type exclusion
and more

**Comment by ggppdk** [comment]
we need to update the next / previous article, navigation field to use current filtering of the content listings instead of current category

since this would be the optimal experience for the frontend user ?

i mention this since your suggestion gets the items in a similar way like FLEXIcontent navigation field, but that is another issue, i ll close this one

**Comment by ggppdk** [comment]
So we can leave this open, 
- just as mention above, we will update our the next / previous article, navigation field, insteading of adding a new field

Work needed for this issue is the same needed by endless scroll feature
- so a milestone 3.2 or later is appropriate

**Comment by ggppdk** [comment]
@Lyquix 

this was implemented in v3.0.x, not only for category view, but for all category based views, closing
